pkg/database/instrumented_rwmutex.go

```diff
@@ -58,15 +54,10 @@ func (imux *instrumentedRWMutex) Unlock() {
 }
 
 func (imux *instrumentedRWMutex) RLock() {
-	imux.trwmutex.Lock()
-	imux.waitingCount++
-	imux.trwmutex.Unlock()
+	atomic.AddInt64(&imux.waitingCount, 1)
+	defer atomic.AddInt64(&imux.waitingCount, -1)
```

Comment:
I do not think that it makes sense to use defer here, just call after rlock.

Reply:
fixed